### ISSUE TITLE
Random errors when using archive releases

### ISSUE THREAD

**Issue by RustyKnight** [Carthage/Carthage#1416]
* `carthage version`:  0.17.2
* `xcodebuild -version`:  Xcode 8.0, Build version 8S162m* Are you using `--no-build`?
* Are you using `--no-use-binaries`? No
* Are you using `--use-submodules`? No

**Cartfile**
```
# Contains the dependencies which are managed as archive releases
git "<internal library>" "develop"

github "jdg/MBProgressHUD" "master"
```

**Internal Library Carfile**
These are maintained as release archives in order to speed up the overall build process
```
github "RustyKnight/CocoaLumberjack" "swift3"

github "RustyKnight/SwiftyJSON" "swift3b2"

github "robbiehanson/CocoaAsyncSocket" "master"

github "RustyKnight/SwiftEventBus" "swift3"

github "realm/realm-cocoa.git" "master"
```

**Carthage Output**
```
*** Fetching MBProgressHUD
*** Fetching ...
*** Fetching SwiftEventBus
*** Fetching realm-cocoa
*** Fetching CocoaAsyncSocket
*** Fetching SwiftyJSON
*** Fetching CocoaLumberjack
*** Checking out ... at "fd777a1ba55923c39f6c683d75b2e99e2f71348a"
*** Downloading CocoaAsyncSocket.framework binary at "XCode8b2-downgrade-2"
*** Downloading SwiftEventBus.framework binary at "xCode8b2 downgrade"
*** Downloading SwiftyJSON.framework binary at "xCode8b2 downgrade"
*** Downloading CocoaLumberjack.framework binary at "Xcode8b2 Downgrade-2"
*** Checking out MBProgressHUD at "59a8240da67a8c98ac7e77da2fa198d21b1ab03c"
*** Downloading realm-cocoa.framework binary at "Xcode8b2 Downgrade-2"
Failed to write to /Users/.../development/.../Carthage/Build/watchOS/CocoaLumberjack.framework.dSYM: Error Domain=NSCocoaErrorDomain Code=4 "The file “DWARF” doesn’t exist." UserInfo={NSSourceFilePathErrorKey=/private/var/folders/xt/tw53km651cjc13kmjky1k0x80000gr/T/carthage-archive.aNQHjd/.../CocoaLumberjack/Carthage/Build/watchOS/CocoaLumberjack.framework.dSYM/Contents/Resources/DWARF, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/Users/.../development/.../Carthage/Build/watchOS/CocoaLumberjack.framework.dSYM/Contents/Resources/DWARF, NSFilePath=/private/var/folders/xt/tw53km651cjc13kmjky1k0x80000gr/T/carthage-archive.aNQHjd/.../CocoaLumberjack/Carthage/Build/watchOS/CocoaLumberjack.framework.dSYM/Contents/Resources/DWARF, NSUnderlyingError=0x7fde8b229a80 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
```

In order to improve the performance of Carthage, I've setup some "release archives" for third party libraries.

Basically, this and some other errors occur from time to time, they are inconsistent and can occur several times in a row or as a once off.  Repeatedly attempting to run `carthage update --platform iOS` will eventually resolve the issue

It's not always the same library which causes the problem, it can be any of the archived releases

**Comment by BernardGatt** [comment]
The latest version of Carthage introduced this issue, i've submitted a PR some time ago, see: https://github.com/Carthage/Carthage/pull/1386

The fix works perfectly but I don't have time to write tests at the moment.

**Comment by ikesyo** [comment]
Thanks for the tip @BernardGatt! So this should be fixed now in Carthage 0.18. Closing.
